test/functional/ui/syntax_conceal_spec.lua

```diff
@@ -0,0 +1,401 @@
+          insert("<r> A region with <r> a nested <r> nested region.</r> </r> </r>\n")
+          screen:expect([[
+            {1:-} a region of text {1:-}                                 |
+            {1:-} a region of text {1:-}                                 |
+            {1:-} A region with {1:-} a nested {1:-} nested region.{1:-}         |
+             {1:-} {1:-}                                                 |
+            ^                                                     |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+            ~                                                    |
+                                                                 |
+          ]])
+        end)
+      end)
+    end) -- Region
+
+    describe("Implicit", function()
+      local syn_region_r_text = "syn region rText start='<r>' end='</r>' conceal cchar=-"
+      local syn_region_i_text = "syn region iText start='<i>' end='</i>' conceal cchar=*"
+      local implicit_conceal_on = "syntax conceal on"
+      local implicit_conceal_off = "syntax conceal off"
+
+      it("Turn on", function()
+        execute(implicit_conceal_on) 
+        feed("2")
+        insert("<r> a region of text </r>\n")
+        execute(syn_region_r_text)
+        screen:expect([[
+          {1:-}                                                    |
+          {1:-}                                                    |
+          ^                                                     |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+          ~                                                    |
+                                                               |
+        ]])
+      end)
+
+      it("Turn On, Off, On", function()
+        execute(implicit_conceal_on) 
+        feed("2")
+        insert("<r> a region of text </r>\n")
+        execute(syn_region_r_text)
+        execute(implicit_conceal_off)
+        feed("2")
+        insert("<i> italian text </i>\n")
+        execute(syn_region_i_text)
```

Comment:
I think there should be a screen test here, to show that definitions are not implicitly concealed.